src/Compilers/Core/Rebuild/BuildConstructor.cs

```diff
@@ -281,23 +268,39 @@ private static VisualBasicCompilationOptions CreateVisualBasicCompilationOptions
             static T? ToEnum<T>(string value) where T : struct => Enum.TryParse<T>(value, out var enumValue) ? enumValue : null;
         }
 
-        public static unsafe EmitResult Emit(
+        public static EmitResult Emit(
             Stream rebuildPeStream,
             CompilationOptionsReader optionsReader,
             Compilation producedCompilation,
             CancellationToken cancellationToken)
         {
-            var peHeader = optionsReader.PeReader.PEHeaders.PEHeader!;
-            var win32Resources = optionsReader.PeReader.GetSectionData(peHeader.ResourceTableDirectory.RelativeVirtualAddress);
-            using var win32ResourceStream = new UnmanagedMemoryStream(win32Resources.Pointer, win32Resources.Length);
-
-            var sourceLink = optionsReader.GetSourceLinkUTF8();
-
             var embeddedTexts = producedCompilation.SyntaxTrees
                     .Select(st => (path: st.FilePath, text: st.GetText()))
                     .Where(pair => pair.text.CanBeEmbedded)
                     .Select(pair => EmbeddedText.FromSource(pair.path, pair.text))
                     .ToImmutableArray();
+            return Emit(
+                rebuildPeStream,
+                optionsReader,
+                producedCompilation,
+                embeddedTexts,
+                cancellationToken);
+        }
+
+        public static unsafe EmitResult Emit(
```

Comment:
Should we make this overload 'internal'?

Reply:
Sure.